model/inline/src/main/scala/aqua/model/inline/raw/ApplyPropertiesRawInliner.scala

```diff
@@ -57,24 +57,87 @@ object ApplyPropertiesRawInliner extends RawInliner[ApplyPropertyRaw] {
         State.pure(vm -> Inline.empty)
     }
 
+  /**
+   * @param valueWithFirstProperty pass value only with the first property to check if we are trying to join a stream
+   */
   private[inline] def unfoldProperty[S: Mangler: Exports: Arrows](
-    p: PropertyRaw
+    p: PropertyRaw,
+    valueWithFirstProperty: Option[ValueModel]
   ): State[S, (PropertyModel, Inline)] = // TODO property for collection
-    p match {
-      case IntoFieldRaw(field, t) =>
+    (valueWithFirstProperty, p) match {
+      case (_, IntoFieldRaw(field, t)) =>
         State.pure(IntoFieldModel(field, t) -> Inline.empty)
-      case IntoIndexRaw(vm: ApplyPropertyRaw, t) =>
+      case (Some(v @ VarModel(_, st @ StreamType(_), _)), IntoIndexRaw(vr, t)) =>
```

Comment:
Could you please explain what's going on there with a comment. E.g. "To wait for the element of a stream by the given index, the following model is generated: ..." with AIR-like snippet of what is generated?

I'm pretty sure we will forget why it is done this way otherwise.